tests/tf-manifests/aws/proxy/main.tf

```diff
@@ -0,0 +1,74 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.20.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+resource "tls_private_key" "proxy_ssh_key" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "generated_key" {
+  key_name   = "imported-proxy-key"
+  public_key = tls_private_key.proxy_ssh_key.public_key_openssh
+}
+
+locals {
+  trust_bundle_path = "${path.module}/files/mitm-ca.pem"
+}
+
+resource "aws_instance" "tf-proxy" {
+  ami           = "ami-05a03e6058638183d"
+  instance_type = "t3.medium"
+  key_name      = aws_key_pair.generated_key.key_name
+  subnet_id     = var.subnet_public_id
+  vpc_security_group_ids = var.security_group_ids
+  associate_public_ip_address = true
+  
+  tags = { Name = "tf-proxy" }
+
+  provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "ec2-user"
+      private_key = "${tls_private_key.proxy_ssh_key.private_key_pem}"
+      host        = "${self.public_ip}"
+      agent       = false  # Disable agent forwarding
+      timeout     = "2m"
+    }
+
+    inline = [
+      "sleep 30",
+      "sudo yum install -y wget",
+      "wget https://snapshots.mitmproxy.org/7.0.2/mitmproxy-7.0.2-linux.tar.gz",
+      "mkdir mitm",
+      "tar zxvf mitmproxy-7.0.2-linux.tar.gz -C mitm",
+      "cd mitm",
+      "nohup ./mitmdump --showhost --ssl-insecure --ignore-hosts quay.io registry.redhat.io amazonaws.com > mitm.log 2>&1 &",
+      "sleep 5",  # Add a delay to allow mitmdump to start
+      # Generate the file locally
+      "http_proxy=127.0.0.1:8080 curl -s http://mitm.it/cert/pem > ~/mitm-ca.pem"
+    ]
+  }
+
+  provisioner "local-exec" {
+        command = <<-EOT
+            echo '${tls_private_key.proxy_ssh_key.private_key_pem}' > /tmp/private_key.pem
+            chmod 600 /tmp/private_key.pem
+            scp -i /tmp/private_key.pem -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ec2-user@${aws_instance.tf-proxy.public_ip}:~/mitm-ca.pem ${path.module}/files/
```

Comment:
Do not use `path.module` to store the CA file. Let's put it to rhcs_output dir. Make the output dir a parameter
```
variable "trust_bundle_path" {
  type        = string
  description = "the file path of the trust bundle"
}
```
And in my code call, I put 
```
	proxyArgs := &EXE.ProxyArgs{
		Region:              region,
		PublicVpcID:         vpcPublicID,
		PublicSubnetID:      subnetPublicID,
		SecurityGroupIDs:    securityGroupIDs,
		TrustBundleFilePath: path.Join(cfg.RhcsOutputDir, "ca.cert"),
	}
```